src/main/java/competition/subsystems/turret/commands/TurretRotationCommand.java

```diff
@@ -0,0 +1,37 @@
+package competition.subsystems.turret.commands;
+
+import com.google.inject.Inject;
+
+import competition.operator_interface.OperatorInterface;
+import competition.subsystems.turret.TurretSubsystem;
+import xbot.common.command.BaseCommand;
+
+public class TurretRotationCommand extends BaseCommand
+{
+    //comment so i can commit this one character change
+    final OperatorInterface oi;
+    final TurretSubsystem turretSubsystem;
+
+    @Inject
+    public TurretRotationCommand(OperatorInterface oi , TurretSubsystem tSubsystem) {
+        this.oi = oi;
+        this.turretSubsystem = tSubsystem;
+        this.addRequirements(this.turretSubsystem);
+    }
+
+    @Override
+    public void execute() {
+        double yAxis = oi.gamepad.getRightStickX();
+        if(yAxis>0 && turretSubsystem.canTurnLeft() || yAxis<0 && turretSubsystem.canTurnRight())
```

Comment:
this should be taken care of by the turretSubsytem right? I think you can remove the logic from here?